pylint/checkers/variables.py

```diff
@@ -2022,18 +2022,19 @@ def _store_type_annotation_names(self, node):
             return
         self._store_type_annotation_node(node.type_annotation)
 
-    def _check_self_cls_assign(self, node):
+    def _check_self_cls_assign(self, node: nodes.Assign) -> None:
         """Check that self/cls don't get assigned"""
-        assign_names = {
-            target.name
-            for target in node.targets
-            if isinstance(target, nodes.AssignName)
-        }
+        assign_names: Set[Union[str, None]] = set()
```

Comment:
`Set` and `Union` need to be added as imports.

Reply:
Changed to use `Optional`. That should be fine as well right? I'll merge if you give the okay!

Reply:
Optional works too.